### PR TITLE
Address Snyk issues

### PR DESCRIPTION
**Pull request by juniortaeza**
### Desired Outcome

This pull request addresses the Snyk issues outlined in this repository.

### Implemented Changes

- upgrades Go to 1.20
- upgrades golang.org/x/net to v0.10.0
- upgrades golang.org/x/text to v0.9.0
- upgrades golang.org/x/sys to v0.8.0

#### Changelog

- [X] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [X] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [X] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [X] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [X] There are no security aspects to these changes

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,8 +12,40 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
-go 1.18
+go 1.19
```
Comment (john-odonnell) [inline go.mod:15]: Project Go version is still 1.19, should be 1.20

File: go.mod [hunk 2]
```diff
@@ -12,8 +12,40 @@ require (
...
+
+replace golang.org/x/sys v0.5.0 => golang.org/x/sys v0.8.0
+
+replace golang.org/x/text v0.3.0 => golang.org/x/text v0.9.0
+
+replace golang.org/x/text v0.3.3 => golang.org/x/text v0.9.0
+
+replace golang.org/x/text v0.7.0 => golang.org/x/text v0.9.0
+
+replace golang.org/x/text v0.3.7 => golang.org/x/text v0.9.0
```
Comment (john-odonnell) [inline go.mod:49]: Currently only v0.3.7 of golang.org/x/text is listed in `go.sum`. Why do we need all these replacements?

Comment (john-odonnell) [reply go.mod:49]: Same goes for x/sys and x/net, I guess.

File: go.mod (hunk 1 above)
Comment (juniortaeza) [reply go.mod:15]: The following error was produced when I tried using 1.20:

`go.mod file indicates go 1.20, but maximum supported version is 1.19`

Comment (john-odonnell) [reply go.mod:15]: You may not have Go 1.20 installed - run `go version` to see what version you're using locally.

File: go.mod (hunk 2 above)
Comment (juniortaeza) [reply go.mod:49]: It looks like after adding these replace statements the go.sum was condensed – the old list showed these versions. Should I remove these replace statements after updating go.sum?

Comment (john-odonnell) [reply go.mod:49]: Are you sure you weren't looking at an amended list? If you look at the `go.sum` file on the `main` branch, [only v0.3.7 of golang.org/x/text is listed](https://github.com/cyberark/summon-aws-secrets/blob/main/go.sum#L22). I wonder if we could get away with only one of these replace statements.

Comment (john-odonnell) [reply go.mod:49]: Good call - makes sense.

**Review by john-odonnell — APPROVED** [review]
LGTM 👍

**Pull request merged**